bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/GoEChargerHandlerFactory.java

```diff
@@ -56,13 +56,23 @@ public boolean supportsThingType(ThingTypeUID thingTypeUID) {
     @Override
     protected @Nullable ThingHandler createHandler(Thing thing) {
         ThingTypeUID thingTypeUID = thing.getThingTypeUID();
-        var apiVersion = thing.getConfiguration().as(GoEChargerConfiguration.class).apiVersion;
+
+        var config = thing.getConfiguration().as(GoEChargerConfiguration.class);
+        var apiVersion = config.apiVersion;
 
         if (THING_TYPE_GOE.equals(thingTypeUID)) {
             if (apiVersion == 1) {
+                if (config.ip == null) {
+                    // apiVersion == 1 requires ip
+                    return null;
+                }
```

Comment:
We usually prefer to do these checks within the handler and set the thing state accordingly  (OFFLINE CONFIGURATION_ERROR ).
Checking API version should be enough to determine the right handler.